pkg/controllers/disruption/controller.go

```diff
@@ -213,7 +213,7 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 	// tainted with the Karpenter taint, the provisioning controller will continue
 	// to do scheduling simulations and nominate the pods on the candidate nodes until
 	// the node is cleaned up.
-	schedulingResults.Record(logging.WithLogger(ctx, operatorlogging.NopLogger), c.recorder, c.cluster)
+	schedulingResults.Record(log.IntoContext(ctx, logging.IgnoreDebugEvents(log.FromContext(ctx))), c.recorder, c.cluster)
```

Comment:
Actually no, but I'm not sure what to supply because `opeatorlogging.NopLogger` is of the type  `*zap.SugaredLogger` while `log.IntoContext` expects `logr.Logger`.

Reply:
I think you can wrap the `Nop` logger with `zapr.NewLogger` like `zapr.NewLogger(zap.NewNop())`

Reply:
Thanks.